packages/engine/paima-sm/src/index.ts

```diff
@@ -591,6 +684,17 @@ async function processUserInputs(
   return latestChainData.submittedData.length;
 }
 
+async function sendEventsToBroker<Event extends EventPathAndDef>(
+  eventsToEmit: EventsToEmit<Event>
+) {
+  if (ENV.MQTT_BROKER) {
+    // we probably don't want to use Promise.all since it will change the order.
+    for (const [eventDefinition, fields] of eventsToEmit) {
+      await PaimaEventManager.Instance.sendMessage(eventDefinition, fields);
```

Comment:
I want to verify something just in case. Currently the emitted events look like this:

```
app/indexedField1/indexedField2 b'{"field1":"data","field2":"data"}'
``

This does seem to be consistent with the builtin events, but I'm not sure if not having the event name or some other identifier makes sense? Since I'm not sure how this would be parsed otherwise, unless a unique indexed field is used explicitly as a tag?

Reply:
TODO: add the topic/signatureHash as part of the prefix (but not for builtin events)